### PR TITLE
Feature/api specs

### PR DESCRIPTION
**Pull request by mtakac2**

### CODE REVIEW

File: spec/api/activity_resource_spec.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    end
+
+    context 'anonymous user' do
+      let(:get_path) { "/api/v3/activities/#{activity.id}" }
+      let(:project) { FactoryGirl.create(:project, is_public: true) }
+      after { Setting.delete_all }
+
+      context 'when access for anonymous user is allowed' do
+        before do
+          Setting.login_required = 0
```
Comment (myabc) [inline spec/api/activity_resource_spec.rb]: @mtakac if you're going to change a setting in a `before` block, you should change it back to the original value in an `after` block (or better, use an `around` block).

File: spec/api/activity_resource_spec.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          get get_path
+        end
+
+        it 'should respond with 200' do
+          expect(subject.status).to eq(200)
+        end
+
+        xit 'should respond with correct activity' do
```
Comment (myabc) [inline spec/api/activity_resource_spec.rb]: @mtakac should this still be pending?

File: spec/api/activity_resource_spec.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        member.save!
+        get get_path
+      end
+
+      it 'should respond with 200' do
+        expect(subject.status).to eq(200)
+      end
+
+      xit 'should respond with correct activity' do
+        expect(subject.body).to be_json_eql(representer.to_json)
```
Comment (myabc) [inline spec/api/activity_resource_spec.rb]: @mtakac IMO these specs are of the integration variety – if you're serialising the representer here and asserting against that, you're not testing the whole stack. You should write explicit assertions for the response (either whole, or partial).

File: spec/api/activity_resource_spec.rb (hunk 1 above)
Comment (myabc) [inline spec/api/activity_resource_spec.rb]: Sorry, I missed the `Setting.delete_all`(I don't know if deleting all Settings is actually necessary though)

File: spec/api/activity_resource_spec.rb (hunk 2 above)
Comment (mtakac2) [inline spec/api/activity_resource_spec.rb]: @myabc yes - it's not done.

File: spec/api/activity_resource_spec.rb (hunk 1 above)
Comment (mtakac2) [inline spec/api/activity_resource_spec.rb]: @myabc Good point - I'll change it to around block.

File: spec/api/activity_resource_spec.rb (hunk 3 above)
Comment (mtakac2) [inline spec/api/activity_resource_spec.rb]: @myabc Okay, point taken. I'll change it.

**Comment by myabc** [comment]
Do we want to merge pending specs?

**Comment by mtakac2** [comment]
@myabc I've already implemented all the pending tests. Let me know if I am wrong but I don't see any pending tests in the PR.

**Comment by myabc** [comment]
@mtakac sorry, you're right. This looks good then.

**Pull request merged**